.toBase64(): was never merged. Users should use their own implementat…

pmlopes opened the pull request:
…ions if needed

Signed-off-by: Paulo Lopes <pmlopes@gmail.com>

Motivation:

Fix #67

sangeetaraghu commented on asciidoc/modules/con_mg_changes-in-json.adoc:
```diff
@@ -107,7 +107,25 @@ If you want to continue using the Base64 encoder in {VertX} {v4}, you can use th
+}
+----
+
+Users should consider using these utility methods for:
```
You must use the utility methods in the following scenarios:

sangeetaraghu commented on asciidoc/modules/con_mg_changes-in-json.adoc:
```diff
@@ -107,7 +107,25 @@ If you want to continue using the Base64 encoder in {VertX} {v4}, you can use th
 
-A new utility method `Json.toBase64()` converts the Base64URL string to Base64. This utility can be for:
+Users that must use both {VertX} {v3x} and {VertX} {v4} releases may need to create some utility to convert from Base64
+to Base64URL. Given that both encodings are Base64, we're just changing the alphabet which makes this a trivial task:
```
@pmlopes Did I get the context right here?

If you want to use both {VertX} {v3x} and {VertX} {v4} releases, you can use the following utility to convert the Base64
string to Base64URL. Since both the encodings are Base64, you will have to make minimal updates to your existing applications.

pmlopes commented:
@sangeetaraghu I've updated the text like you proposed and added another section as feedback from users trying to migrate JSON -> POJO.

Can you do a second check!

Thanks!

sangeetaraghu commented on asciidoc/modules/con_mg_changes-in-json.adoc:
```diff
@@ -118,7 +137,39 @@ Use the following example code to convert a Base64URL to Base64 encoder.
+
+The helper functions described above address plain JSON migrations. If you use automatic object mapping from JSON
+to a {Java} POJO, then a custom object mapper needs to be created to your specific use case.
+
```
@pmlopes Can we document the helper function names? Also again is my understanding correct?

The helper functions XXX in the utility enable only JSON migrations. If you use automatic object mapping from JSON
to a {Java} POJO in your applications, then you must create a custom object mapper to convert the Base64
string to Base64URL.

sangeetaraghu commented on asciidoc/modules/con_mg_changes-in-json.adoc:
```diff
@@ -118,7 +137,39 @@ Use the following example code to convert a Base64URL to Base64 encoder.
+The helper functions described above address plain JSON migrations. If you use automatic object mapping from JSON
+to a {Java} POJO, then a custom object mapper needs to be created to your specific use case.
+
+An object mapper with a custom Base64 decoder:
```
The following example shows you how to create a object mapper with custom Base64 decoder.

sangeetaraghu commented on asciidoc/modules/con_mg_changes-in-json.adoc:
```diff
@@ -118,7 +137,39 @@ Use the following example code to convert a Base64URL to Base64 encoder.
+
+[source,java,options="nowrap",subs="attributes+"]
+----
+// simple deserializer from base64 to byte[]
```
Do we want document it as Base64 or base64?

// simple deserializer from Base64 to byte[]

sangeetaraghu commented on asciidoc/modules/con_mg_changes-in-json.adoc:
```diff
@@ -118,7 +137,39 @@ Use the following example code to convert a Base64URL to Base64 encoder.
+
+ObjectMapper mapper = new ObjectMapper();
+
+// create a custom module to address the base64 decoding
```
Do we want document it as Base64 or base64?

// create a custom module to address the Base64 decoding

sangeetaraghu commented:
@pmlopes Made some minor suggestions.
